p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,23 +1,62 @@
-import React from 'react'
-import { Card } from '../Shared/components/Card'
-import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import React, { useState, useEffect } from "react";
+import { Card } from "../Shared/components/Card";
+import { Button, Input } from "../Shared/components";
+import { fieldsList, isFieldValid, handleSubmitForm } from "./Form.services";
 
-import "./Form.styles.css"
+import "./Form.styles.css";
+const initialState = fieldsList.map((field) => ({
+  ...field,
+  isValid: false,
+  value: "",
+}));
 
 const Form = () => {
-  
+  const [state, setState] = useState(initialState);
+  const [disableButton, setdisableButton] = useState(true);
+
+  useEffect(() => {
+    setdisableButton(state.some((field) => !field.isValid));
+    console.log(state.some((field) => !field.isValid));
+  }, [state]); // Only re-run the effect if state changes
+
+  const handleOnChange = (value, name) => {
+    setState((nextState) => {
+      return nextState.map((field) => {
+        return field.name === name
+          ? { ...field, value: value, isValid: isFieldValid(value) }
+          : field;
+      });
+    });
+  };
+  const handleClearForm = () => {
+    // reset form to initial state
+    setState(initialState);
+  };
+
   return (
     <Card width={"50%"}>
       <div className="form">
-        <Input label="this is label" />
+        {state?.map(({ name, label, type, value }) => (
+          <Input
```

Comment:
this is just a suggestion, and it may depend on eslint configurations, but having the same names for input props and state, you can use a spread operator. example:

```
state?.map(({ isValid, ...inputProps }) => (
  <Input key={inputProps.name} {...inputProps} onChange={handleOnChange} />
));
```

both are completely good way of do it.